### PR TITLE
SDIT-2044 Added person contacts

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/contactperson/ContactPersonResource.kt [hunk 1]
```diff
@@ -219,3 +221,41 @@ data class Identifier(
...
+  val emergencyContact: Boolean,
+  @Schema(description = "Free format comment text")
+  val comment: String?,
+  @Schema(description = "The prisoner this person is a contact for")
+  val prisoner: ContactForPrisoner,
+)
+
+@Schema(description = "The data held in NOMIS about a person's contact with a prisoner")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class ContactForPrisoner(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/contactperson/ContactPersonResource.kt:252]: No idea what is needed yet, but this is useful for debugging endpoint if you know some basic details about the prisoner

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Offender.kt [hunk 2]
```diff
@@ -57,8 +56,8 @@ data class Offender(
   @Column(name = "CASELOAD_TYPE")
   val caseloadType: String? = "INST",
 
-  @Column(name = "FIRST_NAME", nullable = false)
-  val firstName: String? = null,
+  @Column(name = "FIRST_NAME", nullable = true)
+  val firstName: String,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Offender.kt:61]: column is nullable but there is no null data - so switched nullability around

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderContactPerson.kt [hunk 3]
```diff
@@ -35,12 +34,15 @@ data class OffenderContactPerson(
   @JoinColumn(name = "OFFENDER_BOOK_ID", nullable = false)
   val offenderBooking: OffenderBooking,
 
-  @ManyToOne(optional = false, fetch = LAZY)
+  @ManyToOne(optional = true, fetch = LAZY)
   @JoinColumn(name = "PERSON_ID")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderContactPerson.kt:38]: a contact is either related to a civil person or a prisoner - hence this is nullable

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderContactPerson.kt [hunk 4]
```diff
@@ -53,22 +55,15 @@ data class OffenderContactPerson(
   )
   val relationshipType: RelationshipType,
 
-  @ManyToOne
-  @NotFound(action = IGNORE)
+  @ManyToOne(optional = false, fetch = LAZY)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderContactPerson.kt:58]: all pre data is valid - so removed unesscessary NotFound

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsServiceTest.kt [hunk 5]
```diff
@@ -124,7 +124,7 @@ private fun booking(bookingSequence: Int = 1): OffenderBooking =
   OffenderBooking(
     bookingSequence = bookingSequence,
     bookingBeginDate = LocalDateTime.now(),
-    offender = Offender(nomsId = "A1234KT", gender = Gender("M", "MALE"), lastName = "SMITH"),
+    offender = Offender(nomsId = "A1234KT", gender = Gender("M", "MALE"), lastName = "SMITH", firstName = "JOHN"),
   )
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/alerts/AlertsServiceTest.kt:128]: firstName now mandatory

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**